Make expandable list button text more specific

cbliard opened the pull request:
# Ticket

No ticket initially, then scope increased with the feedbacks so this ticket was created to track the changes: https://community.openproject.org/projects/meetings-stream/work_packages/57911/activity

# What are you trying to accomplish?

I don't like this label:

![image](https://github.com/user-attachments/assets/408a145b-c0e0-4775-92a5-e932de2f84bb)

"Show/hide 30 items". That does not sound right. And I suppose screen readers do not read that properly either.
I'd like to display "Show 30 items" when it's hidden, and "Hide 30 items" when it's shown.

I was annoyed by that for a long time, and today I stumbled upon it while reading the lookbook, so why not fix it.

Now there are more to it:
* Change the phrasing of "Show 4 items"/"Hide 4 items" at the end of the participant section with:
  * "Show 4 more" (when collapsed)
  * "Show less" (when expanded)
* Increase margin above the Show link from 4px to 8px
* Remove the text ("Attached files are available...") from below the "Participants" heading
  This text seems to have been introduced as a bug in a prior change (regression)

## Screenshots

Before:

![image](https://github.com/user-attachments/assets/d62efe46-74d1-429f-9a54-d207f4e3d49f)

Now:

![image](https://github.com/user-attachments/assets/c37bdf5d-f5fb-4ddc-b5c7-cef4d1316b8a)

![image](https://github.com/user-attachments/assets/ac6bcdf1-a198-43d4-91b4-b2ba46531418)


# What approach did you choose and why?

There are two buttons. One "Show", one "Hide". Only one is displayed at once. Clicking one changes the one being displayed.

Also fixed the action which did not trigger when pressing "enter". By not passing the action name explicitly, stimulus uses the default action, which should always work better, and is less verbose code-wise.

Also support @cuttoff_limit being nil, in which case all items are displayed (because when changing the value in lookbook, it crashed when I cleared the field before setting a new value).

Also increased to margin of the button from 4px to 8px as agreed in [frontend meeting](https://community.openproject.org/projects/gmbh/meetings/3590).

# Merge checklist

- [ ] Added/updated tests
  - nope, that's just a minor refactoring. I'll update existing tests if some fail.
- [x] Added/updated documentation in Lookbook (patterns, previews, etc)
- [x] Tested major browsers (Chrome, Firefox, Edge, ...)

cbliard committed:
Make expandable list button text more specific

- Instead of "Show/hide", action is now "Show" or "Hide" based on the status.
- Make the translation count-aware.
- Fixed the button action which did not trigger when pressing Enter.
- Support `cutoff_limit` being `nil` and show all elements.
- Changed the margin top of the button from 4px to 8px.

oliverguenther reviewed (approved):
Nice 👏

psatyal commented:
Looks much better, @cbliard. Thank you!

![Screenshot 2024-09-13 at 11 06 24](https://github.com/user-attachments/assets/0dbd69e4-cb3f-4c16-8616-30a95d9a9084)
![Screenshot 2024-09-13 at 11 06 32](https://github.com/user-attachments/assets/18ccaffb-6c41-4217-9e2c-cbc0d2c51f33)

Some feedback:

- The spacing between the users and "Show 4 items" is okay now.
- The term "item" was always an odd choice. Let's change the text to "Show 6 more" and "Show less". (Or simply "Show more" and "Show less")
- The text below "Participants" is wrong; the participant section is not supposed to have text at all. This seems like a regression. Could you please remove it? The spacing also seems too much between the end of the text and the first participant (32px vs. 16px). 

This increases a scope a little bit. I'll create a WP with these requirements if it's okay with you.

Cheers
Parimal

psatyal commented:
Created work package: https://community.openproject.org/projects/meetings-stream/work_packages/57911/activity

cbliard committed:
[57911] Improve participants side panel phrasing and spacing

https://community.openproject.org/wp/57911

* Change the phrasing of "Show 4 items"/"Hide 4 items" at the end of the
  participant section with:
  * "Show 4 more" (when collapsed)
  * "Show less" (when expanded)
* Increase margin above the Show link from 4px to 8px
* Remove the text ("Attached files are available...") from below the
  "Participants" heading, and reduce spacing.

  This text seems to have been introduced as a bug in a prior change

Also removed a level of useless nesting of flex layouts.

bsatarnejad reviewed (approved):
LGTM 👍
